src/featbin/compute-filter.cc

```diff
@@ -0,0 +1,223 @@
+// w.r.t autocorrelation coefficients by minimizing the prediction
+// error using MSE.
+// Durbin used to compute LP coefficients using autocorrelation coefficients.
+// R(j) = sum_{i=1}^P R((i+j % p)] * a[i] j = 0, ..,P
+// P is order of Linear prediction 
+// lp_coeffs - linear prediction coefficients. (predicted_x[n] = sum_{i=1}^P{a[i] * x[n-i]})
+// R(j) is the j_th autocorrelation coefficient.
+void ComputeFilters(const VectorBase<BaseFloat> &autocorr, 
+                    Vector<BaseFloat> *lp_coeffs) {
+  int32 n = autocorr.Dim() - 1;
+  lp_coeffs->Resize(n);
+  // compute lpc coefficients using autocorrelation coefficients. 
+  ComputeLpc(autocorr, lp_coeffs);  
+}
+
+} // namescape kaldi
+
+int main(int argc, char *argv[]) {
+  try {
+    using namespace kaldi;
+    using kaldi::int32;
+
+    const char *usage = 
+      "Computes LP coefficients per-speaker, by minimizing "
+      "prediction error using MSE.\n"
+      "This coefficient contain speaker-dependent information correspond to each speaker.\n"
+
+      "Usage: compute-filters [options] <wave-rspecifier> <filter-rspecifier> \n"
+      "e.g.: compute-filters " 
+      " scp:data/train/wav.scp ark,scp:filter.ark,filter.scp\n";
+
+    ParseOptions po(usage);
+    std::string spk2utt_rspecifier,
+      spk_mean_rspecifier;
+    bool binary = true, 
+      use_mean = false;
+    int32 channel = -1,
+      lpc_order = 100;
+    po.Register("binary", &binary, "write in binary mode (applies only to global filters)");
+    po.Register("lpc-order", &lpc_order, "number of LP coefficients used to extract filters.");
+    po.Register("use-mean-norm", &use_mean, "If true, autocorrelation computed on"
+                " the utterance, which mean normalized using mean computed per speaker." 
+                "If false, the utterance is mean-normalized w.r.t its local mean.");
+    po.Read(argc, argv);
+    
+    if (po.NumArgs() != 2) {
+      po.PrintUsage();
+      exit(1);
+    }
+    int32 num_done = 0, num_err = 0;
+    std::string wav_rspecifier = po.GetArg(1),
+      wspecifier = po.GetArg(2);
+    
+    BaseFloatVectorWriter writer(wspecifier);
+    if (spk2utt_rspecifier != "") { 
+      SequentialTokenVectorReader spk2utt_reader(spk2utt_rspecifier);
+      RandomAccessTableReader<WaveHolder> wav_reader(wav_rspecifier);
+      for (; !spk2utt_reader.Done(); spk2utt_reader.Next()) {
+        std::string spk = spk2utt_reader.Key();
+        const std::vector<std::string> &uttlist = spk2utt_reader.Value();
+        Vector<BaseFloat> inner_prod(lpc_order),
+          norms(lpc_order);
+        int32 num_chan = wav_reader.Value(uttlist[0]).Data().NumRows();
+        std::vector<BaseFloat> spk_sum(num_chan, 0),
+          spk_count(num_chan, 0),
+          spk_mean(num_chan, 0);
+        // Accumulate waveform sum and count over all utterance in speaker
+        // to compute mean of waveform for whole speaker over all channels.
+        if (use_mean) {
```

Comment:
no reason to compute the mean for each channel, since computation is done only for one channel.
Work out the mean inside the 'AccStatsForCorrelation' function instead.